src/traverser.js

```diff
@@ -17,7 +17,7 @@ class Traverser {
     this.initialPath = initialPath
     this.excludes = excludes
     this.dest = dest
-    this.src = src
+    this.src = src.split('.')
```

Comment:
maybe we conditionally do the split `this.src = src.includes('.') ? src.split('.') : src`  then in `handleFile` check if `Array.isArray(src)` and do logic check if `true` and do the old check if `false`. I think this would make it clearer when looking back at this code at a later date

Reply:
Could be - do you think it'd be enough if I add a comment? I must admit I'm very prone to premature optimization. However only having one flow of logic can be beneficial as well and I think the idea is easy to understand.
But I just wondered: How about not splitting the file name nor `src` at all? In the end all we want to check for is the last text sequence of the file name to match the `src `string. See beneath for possible implementation.